test/url_checker.py

```diff
@@ -90,38 +93,57 @@ def find_invalid_manifest_urls(allowlisted_urls):
   """
   dc_api_key = os.environ.get('DC_API_KEY', '')
 
-  resp = requests.get(
-      API_PREFIX + dc_api_key +
-      "&nodes=Provenance&nodes=Dataset&nodes=Source&property=<-typeOf")
-  if resp.status_code != 200:
-    raise ValueError(API_ERROR.format(resp.status_code))
-
+  nextToken = ""
   dcids = []
-  for _, data in resp.json()["data"].items():
-    for node in data["arcs"]["typeOf"]["nodes"]:
-      dcids.append(node["dcid"])
-
-  resp = requests.get(API_PREFIX + dc_api_key +
-                      "".join(["&nodes=" + dcid for dcid in dcids]) +
-                      "&property=->[" + ",".join(URL_PROPS) + "]")
-  if resp.status_code != 200:
-    raise ValueError(API_ERROR.format(resp.status_code))
-
+  while True:
```

Comment:
for future, we could consider a helper:

`def get_with_pagination(url, process_fn):`

Where `process_fn` is a function that takes `resp.json()` and returns nothing.

Reply:
Good idea! I'll do this as a follow up